Realm/Realm.Unity/Editor/Extensions/MethodReferenceExtensions.cs

```diff
@@ -49,4 +49,24 @@ public static MethodReference MakeHostInstanceGeneric(this MethodReference @this
 
         return reference;
     }
+
+    public static bool HasSameNameAs(this MethodReference @this, MethodReference other)
+    {
+        if (@this is null || other is null)
+        {
+            return false;
+        }
+
+        return @this.FullName == other.FullName;
+    }
+
+    public static bool ConstructsSameAs(this MethodReference @this, MethodReference other)
```

Comment:
I'll double check to make sure, but while looking on the web I found that the given examples were looking at FullName of methods to compare their equality.

Reply:
Comparing the method reference with == doesn't return the expected result. Hence, I updated `ConstructsSameAs` with a name check. However, I used the `Name` property and not `FullName` one because fullname brings the whole signature with it. We don't really care with what constructor the object is instantiated, we just care that an object of that type is instantiated.